builder/assignmentEnvImageBuilder.go

```diff
@@ -19,36 +21,41 @@ import (
 	"strings"
 )
 
-type baseEnvImage interface {
-	verifyAndWriteInstructions() error
-	build() error
-	publish() error
-	resetDockerfileData()
-	deleteDockerfile() error
-	undoBuild() error
-}
-
-type assignmentEnvironment struct {
+// assignmentEnvironmentImageBuilder is used to build an image with an environment
+// that satisfies all the language and library dependencies of an assignment.
+//
+// The image is built in phases to allow for rollback in situations where the execution of a phase fails.
+// Once the configuration passes a verification phase, a Dockerfile is created and
+// the docker client (here - https://github.com/moby/moby/tree/master/client) is used to generate the image.
+// The image is then pushed to docker hub repository, if required.
+type assignmentEnvironmentImageBuilder struct {
 	DockerfileInstructions bytes.Buffer
 	ImgBuildConfig         *imageBuildConfig
 	AssgnEnvConfig         *configurations.AssignmentEnvConfig
 	ImageExists            bool
 }
 
-type assignmentEnvironmentOption func(*assignmentEnvironment) error
+// assignmentEnvironmentOption is a function interface that is supplied as different
+// options while creating new instance of 'assignmentEnvironmentImageBuilder' type.
+// This function returns any error encountered.
+type assignmentEnvironmentOption func(*assignmentEnvironmentImageBuilder) error
 
-func newAssignmentEnvironment(options ...assignmentEnvironmentOption) (*assignmentEnvironment, error) {
-	assgnEnv := &assignmentEnvironment{}
+// newAssignmentEnvironment takes one or more options and returns new instance
```

Comment:
rewrite this as shown below. You will also need to rename the constructor to `newAssignmentEnvironmentImageBuilder`.
```go
// newAssignmentEnvironmentImageBuilder constructs an instance of assignmentEnvironmentImageBuilder
// by applying each of the provided options.
// The construction of the object fails upon the failure of at least one of the given options.

Reply:
Done